crates/ecs/src/lib.rs

```diff
@@ -375,7 +375,7 @@ pub type ArchetypeResult<T, E = ArchetypeError> = Result<T, E>;
 
 /// Stores components associated with entity ids.
 #[derive(Debug, Default)]
-struct Archetype {
+pub struct Archetype {
```

Comment:
Why `pub`?

Reply:
To reach Archetype in systems/mod.rs, the system parameter Entity needs to save the archetypes it has borrowed to find the entities.

Reply:
I see, then it's preferable to make it crate-private!

```suggestion
pub(crate) struct Archetype {
```

Reply:
Had to revert since there was one more thing that I forgot: Archetype is used in Entity's interface which is public and therefore needs to be public. Is it ok that Archetype is public or do we want to avoid it?

Reply:
It's not a dealbreaker, just something that'd be nice to avoid (ideally, archetypes are an implementation-detail, entirely hidden from outside view). If it's not possible to solve by making the other stuff private as well, then you can just leave it as-is.

Reply:
it is here that it is public: 
```rust
impl SystemParameter for Entity {
    type BorrowedData<'archetypes> = (
        BorrowedArchetypeIndex,
        Vec<(ComponentIndex,&'archetypes Archetype)>,
    );
```

Basically, the systam parameter saves the archetypes in a vector when it does borrow so that it can iterate through them when looking for entity. I can look into it, but I think it's needen.

Reply:
I thought I did? Hmm

Reply:
try running `pre-commit run --all-files` and see if any files are modified